recirq/qml_lfe/util.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Misc utility functions."""
+
+from typing import List
+import os
+import cirq
+import cirq_google
+
+
+def flatten_circuit(circuit: cirq.Circuit) -> cirq.Circuit:
+    """Pack operations in circuit to the left as far as possible.
+
+    Args:
+        circuit: `cirq.Circuit` who's operations will be packed.
+
+    Returns:
+        A `cirq.Circuit` with operations packed to the left as
+        far as possible.
+    """
+    return cirq.Circuit([op for mom in circuit for op in mom])
+
+
+def engine_sim_workaround(
+    batch: List[cirq.Circuit], n_shots: int, use_engine: bool
+) -> List[cirq.Result]:
+    """Use either simulator or engine with a uniform return type.
+
+    Args:
+        batch: List of `cirq.Circuit`s to execute.
+        n_shots: Number of shots to draw from all circuits.
+        use_engine: Whether or not to make use of quantum engine and the
+            weber processor. Note this requires the GOOGLE_CLOUD_PROJECT
+            environment variable to be set, along with the required cloud
+            permissions.
+    """
+    if use_engine:
+        project_id = os.environ["GOOGLE_CLOUD_PROJECT"]
+        engine = cirq_google.Engine(project_id=project_id)
```

Comment:
use `cg.get_engine` which has this logic automatically

Reply:
Done.